apacheconfig/__init__.py

```diff
@@ -6,16 +6,24 @@
 from apacheconfig.lexer import make_lexer
 from apacheconfig.parser import make_parser
 from apacheconfig.loader import ApacheConfigLoader
+from apacheconfig.wloader import ApacheConfigWritableLoader
 from apacheconfig.error import ApacheConfigError
+from apacheconfig import flavors  # noqa: F401
 
 
 @contextmanager
-def make_loader(**options):
+def make_loader(writable=False, **options):
     ApacheConfigLexer = make_lexer(**options)
     ApacheConfigParser = make_parser(**options)
 
-    yield ApacheConfigLoader(ApacheConfigParser(ApacheConfigLexer()),
-                             **options)
+    if writable:
+        options["preservewhitespace"] = True
+        yield ApacheConfigWritableLoader(
+                ApacheConfigParser(ApacheConfigLexer(), start='contents'),
```

Comment:
Possibly a silly question to ask: do we need `ApacheConfigWritableLoader` and `ApacheConfigLoader`? Can we get away with just `ApacheConfigWritableLoader` acting as the only loader (perhaps named just `ApacheConfigLoader`)?

Reply:
Like I mentioned, I think it would be good to keep them separate for the time being, then slowly migrate all the functionality of the original into the new one. However, I suppose I could gate the behavior behind a flag, like we have `writable` in `make_loader`.

Reply:
My only concern is that if we make a release with separate `ApacheConfigWritableLoader`, we'd probably have to deprecate it going forward. In that light it might be cleaner to not leak it into a release.

But we are good to merge this PR regardless!

Reply:
On another thought-- I still think they should be separate for the time being, since the return-types of `load` are different between the two implementations (`dict` vs `ListNode`).

Reply:
Right, may be we could work that around by teaching AST objects generating Python types - then we could emulate old `ApacheConfigLoader` behavior in a thin wrapper calling `ApacheConfigWritableLoader` and then dumping dicts...

My hope is that code duplication would be lesser...

Reply:
I think that makes sense. I'll merge the two for now and make sure that the `writable` flag is properly documented (that it will change the return and argument-types of `load`/`dump`)